### DISCUSSION TITLE
Drawing ascii 0xA

### DISCUSSION THREAD

**Discussion by ZX-80** [python-pillow/Pillow#6568] [Q&A] [answered]
I have some code that takes a few fonts from [INT10h.org](https://int10h.org/oldschool-pc-fonts/) and outputs them as 64x4 character images

```python3
from PIL import Image, ImageDraw, ImageFont

FONT_NAME = "Px437_EverexME_7x8.ttf"
FONT_SIZE = 8
ROWS = 4
COLUMNS = 256 // ROWS

font = ImageFont.truetype(FONT_NAME, FONT_SIZE, encoding='utf-8')
def draw_bitmap() -> Image:
    for row in range(ROWS):
        text = ''.join(chr(i + COLUMNS * row) for i in range(COLUMNS))
        if row == 0:
            _, _, width_image, height_row = font.getbbox(text)
            image = Image.new("RGB", (width_image, height_row * ROWS), "black")
            draw = ImageDraw.Draw(image)
        draw.text((0, height_row * row), text, "white", font, allow_multiline=False)
    return image

draw_bitmap().save(f"{FONT_NAME}.png")
```
But because ascii 10 is being interpreted as a newline, I get this:

![broken](https://user-images.githubusercontent.com/44975876/189393094-a41bf310-54c9-479d-9a77-c9bff112ded5.png)

The docs for the text argument in `ImageDraw.text()` state:
> If it contains any newline characters, the text is passed on to `multiline_text()`

How do I bypass this and force it to ignore the newline? I feel like I'm missing something really simple. For now, I've resorted to modifying `ImageDraw.py` and adding an `allow_multiline` argument that skips the `self._multiline_check(text)` when false, but that's obviously far from ideal:

![working](https://user-images.githubusercontent.com/44975876/189395021-04e3abff-27c8-40ee-87cc-9399690ceec7.png)

**Comment by nulano** [comment, marked as answer]
There is currently no way to do this with `ImageDraw`, and I'm not sure how much interest there would be in adding support for it.

I'd suggest you just copy the relevant code from `ImageDraw.text`:

```python
        def text_force_singleline(draw, xy, text, font, fill, **kwargs):
            class FakeBitmap:
                def load():
                    pass
            bitmap = FakeBitmap()
            mask, offset = font.getmask2(text, "L", **kwargs)
            bitmap.im = mask
            coord = xy[0] + offset[0], xy[1] + offset[1]
            draw.bitmap(coord, bitmap, fill)
```

(Note that your use of `font.getbbox` does not actually give you the size, but that is probably fine with your font. For your specific use-case you might be interested in using `width, height = mask.width, mask.height` and `coord = (0, height * row)`, i.e. ignoring the `offset`, but this might not be helpful for other fonts.)

**Comment by AnonymouX47** [comment]
I think it's much easier to omit `0xA` from the string.
